src/features/comments/commentsSlice.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable no-param-reassign */
+
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Comment } from '../../types/Comment';
+import { getPostComments } from '../../api/comments';
+
+export const fetchComments = createAsyncThunk<Comment[], number>(
+  'comments/fetchComments',
+  postId => {
+    const response = getPostComments(postId);
+
+    return response;
+  },
+);
```

Comment:
It seems like the `fetchComments` async thunk is not handling the promise returned by `getPostComments` correctly. You should await the response or use `.then` to ensure you are returning the resolved data, not the promise itself. This is a bug that will affect the functionality as it won't properly wait for the comments to be fetched before dispatching the action.